patches/picolibc-HEAD.patch

```diff
@@ -15,3 +39,65 @@ index 4bb20f3af..cf9e564d1 100644
  conf_data.set('_HAVE_CC_INHIBIT_LOOP_TO_LIBCALL',
  	      cc.has_argument('-fno-tree-loop-distribute-patterns'),
  	      description: 'Compiler flag to prevent detecting memcpy/memset patterns')
+@@ -1271,9 +1281,16 @@ if has_semihost
+   else
+     custom_script = cpu_family_script
+   endif
+-  test_link_args = ['-nostartfiles',
++  if clang_build
++    clang_libs_list = meson.get_cross_property('clang_libs_list',[])
++    test_link_args = ['-nostartfiles',
++                    '-T', meson.current_source_dir() / custom_script,
++                    '-T', '@0@'.format(picolibc_ld)] + clang_libs_list + additional_libs_list
++  else
++    test_link_args = ['-nostartfiles',
+                     '-T', meson.current_source_dir() / custom_script,
+                     '-T', '@0@'.format(picolibc_ld), '-lgcc'] + additional_libs_list
++  endif
+   test_linker_files = files(custom_script) + [picolibc_ld]
+   specs = picolibc_specs
+ else
+diff --git a/picolibc.ld.in b/picolibc.ld.in
+index c757f7665..f5f794b49 100644
+--- a/picolibc.ld.in
++++ b/picolibc.ld.in
+@@ -175,7 +175,7 @@ SECTIONS
+ 	PROVIDE( __tdata_source_end = LOADADDR(.tdata) + SIZEOF(.tdata) );
+ 	PROVIDE( __data_source_end = __tdata_source_end );
+ 	PROVIDE( __tdata_size = SIZEOF(.tdata) );
+-	PROVIDE( __tls_align = MAX(ALIGNOF(.tdata),ALIGNOF(.tbss)) );
++	/*PROVIDE( __tls_align = MAX(ALIGNOF(.tdata),ALIGNOF(.tbss)) );*/
+ 
+ 	PROVIDE( __edata = __data_end );
+ 	PROVIDE( _edata = __data_end );
+@@ -191,7 +191,7 @@ SECTIONS
+ 	} >ram AT>ram :tls :ram
+ 	PROVIDE( __bss_start = ADDR(.tbss));
+ 	PROVIDE( __tbss_start = ADDR(.tbss));
+-	PROVIDE( __tbss_offset = ADDR(.tbss) - ADDR(.tdata) );
++	/*PROVIDE( __tbss_offset = ADDR(.tbss) - ADDR(.tdata) );*/
+ 	PROVIDE( __tbss_size = SIZEOF(.tbss) );
+ 	PROVIDE( __tls_size = __tls_end - __tls_base );
+ 	PROVIDE( __tls_align = MAX(ALIGNOF(.tdata), ALIGNOF(.tbss)) );
+@@ -208,7 +208,7 @@ SECTIONS
+ 	 * so we create a special segment here just to make room
+ 	 */
+ 	.tbss_space (NOLOAD) : {
+-		. = ADDR(.tbss);
++		/*. = ADDR(.tbss);*/
```

Comment:
If you rebase your patch on the latest version of LLVM-embedded-toolchain-for-Arm I think you'll find this is unnecessary.

Reply:
Thanks.